shelley/chain-and-ledger/dependencies/non-integer/doc/non-integer-calculations.tex

```diff
@@ -75,124 +85,163 @@
-
-These arbitrarily long numerators and denominators incur
-non-predictable~\footnote{potentially > 1s for basic arithmetic operations}
+arithmetic using a pait of numbers to represent a numerator and denominator.
+Effectively, \emph{gcd} calculations are used to normalize the results
+after each basic operation. While not all real numbers are rationals, this approach allows for
+arbitrarily precise approximation of all real numbers. However, arbitrarily
+long numerators or denominators may be needed to represent irrational numbers to the required precision.
+These arbitrarily long numerators and denominators are undesirable, since they will incur non-predictable~\footnote{potentially > 1s for basic arithmetic operations}
 run-time.
 
 \subsection{Fixed Point Arithmetic}
-\label{sec:fixed-point-arithm}
 
-An alternative approach is to use fixed point arithmetic based on exact integers
-where a proportion of the integer is used as fractional part. This prevents the
-problem with using fully rational arithmetic which can get very inefficient and
-allows to define the desired number of decimal digits.
+Fixed point\khcomment{Added this discussion} arithmetic uses a fixed size value (typically 32 or 64 bits) to
+represent non-integer numbers.  A fixed scaling factor is used to determine the
+position of the decimal point and normal integer arithmetic operations are
+applied for addition, multiplication etc.  Like floating point, this has the
+advantage of allowing a compact number representation, and allows fast implementation even on
+a system without floating point supprt.  It also allows precise arithmetic.  However,
+arbitrarily large numbers cannot be supported, which would make it unsuitable for
+representing currency in Cardano (it might, however, be suitable for e.g. leader election calculations).
 
-As it is implemented purely in SW on top of exact integer arithmetic, on the one
-hand it is less efficient than IEEE~754, on the other hand it can be made to
-behave equivalently in different implementations.
 
+\subsection{Fixed Precision Arithmetic}
+\label{sec:fixed-point-arithm}
+
+An alternative approach is to use fixed precision\khcomment{Changed Fixed Point to Fixed Precision} arithmetic based on exact integers
+where a proportion of the integer is used as fractional part.
+The number of fractional digits is fixed to a constant.
+This is more efficient
+than using the rational number approach described above, since it is not necessary to normalise
+values after calculations, arbitrary precision results are not supported, and only a single
+arbitrarily sized integer needs to be stored.
+Since it is implemented purely in software on top of exact integer arithmetic, this approach
+will be less efficient than IEEE~754 floating point.
+However, it can be made to behave equivalently in different implementations.\khcomment{This is not clear: performance, accuracy, ...}
```

Comment:
maybe we could write about what we mean more precisely, i.e., consistent time within each architecture, golden tests to validate consistent results for different architectures